solidity/test/l2/L2TBTC.test.ts

```diff
@@ -0,0 +1,1165 @@
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should add address as a guardian", async () => {
+          // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+          expect(await token.isGuardian(guardian.address)).to.be.true
+        })
+
+        it("should emit an event", async () => {
+          await expect(tx)
+            .to.emit(token, "GuardianAdded")
+            .withArgs(guardian.address)
+        })
+      })
+
+      context("when address is a guardian", () => {
+        before(async () => {
+          await createSnapshot()
+
+          await token.connect(governance).addGuardian(guardian.address)
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should revert", async () => {
+          await expect(
+            token.connect(governance).addGuardian(guardian.address)
+          ).to.be.revertedWith("This address is already a guardian")
+        })
+      })
+
+      context("when there are multiple guardians", () => {
+        const guardians = [
+          "0x54DeA8194aaF652Cd296B162A2809dd95529f775",
+          "0x575E6d8802e7b6A7E8F940640804385D8Bbe2ce0",
+          "0x66ac131D339704902aECCaBDf55e15daAE8B238f",
+        ]
+
+        before(async () => {
+          await createSnapshot()
+
+          await token.connect(governance).addGuardian(guardians[0])
+          await token.connect(governance).addGuardian(guardians[1])
+          await token.connect(governance).addGuardian(guardians[2])
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should add them into the list", async () => {
+          expect(await token.getGuardians()).to.deep.equal(guardians)
+        })
+      })
+    })
+  })
+
+  describe("removeGuardian", () => {
+    context("when called not by the owner", () => {
+      it("should revert", async () => {
+        await expect(
+          token.connect(thirdParty).removeGuardian(guardian.address)
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when called by the owner", () => {
+      context("when address is not a guardian", () => {
+        it("should revert", async () => {
+          await expect(
+            token.connect(governance).removeGuardian(thirdParty.address)
+          ).to.be.revertedWith("This address is not a guardian")
+        })
+      })
+
+      context("when address is a guardian", () => {
+        let tx: ContractTransaction
+
+        before(async () => {
+          await createSnapshot()
+
+          await token.connect(governance).addGuardian(guardian.address)
+          tx = await token.connect(governance).removeGuardian(guardian.address)
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should take guardian role from the address", async () => {
```

Comment:
Can we check the `guardians` array too?

Reply:
205c26e348d81348278fed14debeda8c75a82cd7